pkg/asset/cluster/vsphere/vsphere.go

```diff
@@ -13,3 +16,10 @@ func Metadata(config *types.InstallConfig) *vsphere.Metadata {
 		Password: config.VSphere.Password,
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, infraID string, installConfig *installconfig.InstallConfig) error {
+	// TODO: create VM Template using cachedImage
+	return nil
+}
```

Comment:
the goal for pre terraform was to put in code that was not easily done by terraform itself, like tagging subnets for AWS.

my preference would be keep any cloud creation into terraform, we can easily keep local terraform providers for the custom objects/operations like the uploading ova.